### ISSUE TITLE
swifter install is stuck 

### ISSUE THREAD

**Issue by amankagarwal** [jmcarpenter2/swifter#146]
Hey guys, using Python 3.9 here on my local (MacOS). Tried a simple pip install swifter in my venv, and have not been able to pass through this:

`INFO: pip is looking at multiple versions of ipykernel to determine which version is compatible with other requirements. This could take a while.
Collecting ipykernel>=4.5.1
  Using cached ipykernel-5.4.1-py3-none-any.whl (119 kB)
  Using cached ipykernel-5.4.0-py3-none-any.whl (119 kB)
  Using cached ipykernel-5.3.4-py3-none-any.whl (120 kB)
  Using cached ipykernel-5.3.3-py3-none-any.whl (120 kB)
  Using cached ipykernel-5.3.2-py3-none-any.whl (120 kB)
  Using cached ipykernel-5.3.1-py3-none-any.whl (120 kB)
  Using cached ipykernel-5.3.0-py3-none-any.whl (119 kB)`
  
  I believe someone else also had this issue and has documented it in this stack overflow post.
  https://stackoverflow.com/questions/65238819/failed-to-install-swifter-via-pip-info-pip-is-looking-at-multiple-versions

**Comment by bscholer** [comment]
What I did to get it to install is first install swifter with no dependencies, then go through one by one and install them.

`pip install swifter --no-deps`

It's a bit of a pain, but it keeps it from running for ages (mine ran for about two days before I gave up on it)

**Comment by crimsoneer** [comment]
I had a similar issue...not sure what is causing it

**Comment by Yamp** [comment]
This is probably an issue with dependencies in setup.py. The allowed range probably should be narrowed down. Pip is probably stuck in dependency resolution stage and is trying to download all version one by one.

**Comment by keegean** [comment]
The reason for the install getting stuck is due to the dependency of modin[ray]. The version of modin[ray] does not install or work for Windows. This may be fine for Linux, but if you remove the modin[ray] dependency then swifter installs just fine. The other solution is to have the setup.py check for OS version and change from modin[ray] to modin[dask] if windows is detected.

**Comment by crypdick** [comment]
Had a similar issue to @keegean:

```
Could not find a version that matches ray[serve,tune]<1.2.0,==1.2.0,>=1.0.0 (from -r src/requirements.in (line 49))
Tried: 0.8.4, 0.8.5, 0.8.6, 0.8.7, 1.0.0, 1.0.1, 1.0.1.post1, 1.1.0, 1.2.0
Skipped pre-versions: 1.0.0rc0, 1.0.0rc1, 1.0.0rc2
There are incompatible versions in the resolved dependencies:
  ray[serve,tune]==1.2.0 (from -r src/requirements.in (line 49))
  ray<1.2.0,>=1.0.0 (from modin[ray]==0.9.1->swifter==1.0.7->-r src/requirements.in (line 60))
```

Fixed the issue by switching from swifter directly to Modin.

**Comment by lukleu** [comment]
had the same issue, i ran install with --no-deps and installed the dependencies manually, took a few mins but worked fine

**Comment by jmcarpenter2** [comment]
Hey everyone, thanks so much for the documentation for everyone. This is very helpful. I am planning to make the modin dependency optional, per #147 and #150

**Comment by Biber76** [comment]
> What I did to get it to install is first install swifter with no dependencies, then go through one by one and install them.
> 
> `pip install swifter --no-deps`
> 
> It's a bit of a pain, but it keeps it from running for ages (mine ran for about two days before I gave up on it)

I'm a beginner. How do I know what I have to install after the above command? Can you please guide? Thank you, appreciate the help!

**Comment by bscholer** [comment]
@Biber76 What I did is run that, then try to import the library into my program with `import swifter`, and then use it somewhere with `df.swifter.apply()`. It will likely complain about a package, and say something like `some-package isn't installed`. When that happens, just run `pip install some-package` until it eventually works! A bit of a hassle, but I'm pretty sure this is really an issue with pip and not swifter. Good luck, and happy development!

**Comment by Biber76** [comment]
> @Biber76 What I did is run that, then try to import the library into my program with `import swifter`, and then use it somewhere with `df.swifter.apply()`. It will likely complain about a package, and say something like `some-package isn't installed`. When that happens, just run `pip install some-package` until it eventually works! A bit of a hassle, but I'm pretty sure this is really an issue with pip and not swifter. Good luck, and happy development!

Thank you very much!

**Comment by jmcarpenter2** [comment]
Resolved in swifter==1.0.8 #156

Please refer to the README on how to install swifter without modin (`pip install swifter==1.0.8`)
